cli/src/pcluster/config/cluster_config.py

```diff
@@ -116,23 +351,29 @@ class HeadNodeConfig(Config):
     def __init__(
         self,
         instance_type: str,
-        networking_config: HeadNodeNetworkingConfig,
-        ssh_config: SshConfig,
-        image_config: ImageConfig,
+        networking: HeadNodeNetworkingConfig,
+        ssh: SshConfig,
+        image: ImageConfig = None,
+        storage: StorageConfig = None,
+        dcv: DcvConfig = None,
     ):
         super().__init__()
         self.instance_type = instance_type
-        self.image_config = image_config
-        self.networking_config = networking_config
-        self.ssh_config = ssh_config
-        self._register_validator(InstanceTypeValidator, priority=100, instance_type=self.instance_type)
+        self.image = image
+        self.networking = networking
+        self.ssh = ssh
+        self.storage = storage
+        self.dcv = dcv
+        self._register_validator(InstanceTypeValidator, priority=1, instance_type=self.instance_type)
```

Comment:
not related to your patch but I'd rename _register_validator to _add_validator to be more compact. You can do it with IDE refactor utility.

Reply:
Done